src/frame.js

```diff
@@ -0,0 +1,24 @@
+class Frame {
+  constructor() {
+    this.max_pins = 10
+    this.score = [];
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  rollPins(pins) {
+    if (this.frameOver()) throw new Error('cannot roll more than twice'); 
+    if (pins < 0) throw new Error('cannot roll negative amount of pins');
+    if (pins > this.max_pins) throw new Error('cannot roll more than a maximum amount of pins');
+    if (pins > this.max_pins - this.score[0]) throw new Error('the roll input exceeds the amount of pins left');
```

Comment:
Lovely use of errors. I couldn't find a working way to test for errors so looking forward to seeing your test suite